bokehjs/src/coffee/models/widgets/data_table.coffee

```diff
@@ -99,6 +99,11 @@ export class DataTableView extends WidgetView
     @grid.invalidate()
     @grid.render()
 
+    # XXX: Workaround for `@model.source.trigger('change')` not triggering an event within python.
+    # But we still need it to trigger render updates
+    @model.source.data = @model.source.data
```

Comment:
I see why this works. This line won't emit change signals, because of `isEquals()` check (setting the same data), but server notification is done regardless of that. Lets leave this as-is, but change the comment to indicate that this is only needed to call `@_tell_document_about_change()`.